### DISCUSSION TITLE
A Step-by-Step Guide to Self-Hosting Decap CMS without Netlify | njfamirm

### DISCUSSION THREAD

**Discussion by giscus** [njfamirm/blog#214] [Blog]
# A Step-by-Step Guide to Self-Hosting Decap CMS without Netlify | njfamirm

Simple guide to host Decap CMS backend on your own server.

https://www.njfamirm.ir/en/blog/self-hosting-decap-cms/

**Comment by njfamirm** [comment]
if have any question, ask them here ...

**Reply by mattdelashaw** [reply]
but why Telegram? lol jk i guess that serves it purpose. Nice project

**Comment by snmbala** [comment]
Hi Njfamirm, thanks for the tutorial. I have got mostly correct. But got stuck in callback url. I'm seeing a blank screen after successful login. My Decap is running on 

[![Screenshot](https://i.ibb.co/2KVZ0Br/Screenshot-2024-04-12-at-8-19-07-PM.png)](https://ibb.co/GPYHfjD)

I have installed decap and the oauth in the same server but in different port, oauth on 5555 and the decap on 3333. Not sure if that is the problem

here is my config code [![Screenshot](https://i.ibb.co/98DbY8P/Screenshot-2024-04-12-at-5-16-11-PM.png)](https://ibb.co/YRCcjRg)

**Reply by njfamirm** [reply]
Hi @snmbala, you welcome 🙌🏻
I have not used Decap CMS for a long time...
If you provide me with more information including code or log error I may be able to help.

**Reply by phuhh98** [reply]
Hi
the message post in [decap-cms-github-backend](https://github.com/njfamirm/decap-cms-github-backend/blob/main/src/route/callback.ts#L50) script seem to be sending incorrect payload

The payload for it should be like this [netlify-cms-github-oauth-provider
](https://github.com/vencax/netlify-cms-github-oauth-provider/blob/master/login_script.js#L38) an JSON for object of token and provider values [here](https://github.com/vencax/netlify-cms-github-oauth-provider/blob/master/callback.js#L20)

(@snmbala)[https://github.com/snmbala] you could try this (solution)[https://github.com/vencax/netlify-cms-github-oauth-provider], be aware that ORIGIN env should include base url of your oauth backend.

**Comment by njfamirm** [comment]
New release published now, thanks @czepiec
